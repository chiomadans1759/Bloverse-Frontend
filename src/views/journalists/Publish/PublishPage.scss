#publish-page {
  background: #fff;
  height: auto;
  min-height: 100vh;
  .form-control {
    border: none;
    background: #fff;
    transition: none !important;
    margin-bottom: 2rem;
    color: #252525;
    font-weight: 600;
    &:focus {
      outline: none;
      box-shadow: none;
    }
  }
  .one::placeholder {
    font-weight: 600;
    font-size: 1.25rem;
    color: #868686;
  }
  .two {
    &::placeholder {
      color: #868686;
    }
    > input::placeholder {
      font-weight: 600;
      color: #868686;
    }
  }
  .mytext {
    overflow: hidden;
    width: 100%;
    border: 0;
    margin-right: 1.5rem;
    background: #fff;
    color: #565656;
    resize: none;
    padding-bottom: 3rem;
    font-size: 1rem;
    font-weight: 500;
    &:focus {
      outline: none;
      box-shadow: none;
    }
  }

  .back-icon {
    color: #000000;
  }
  .publish-nav {
    height: 3.5rem;
  }
  .preview {
    border: 0.063rem solid #096DD9;
    color: #096DD9;
    background: transparent;
  }

  .click {
    cursor: pointer;
  }

  .text-icons {
    list-style-type: none;
    margin-top: 4.5rem;
    position: fixed;
    li {
      margin-bottom: 1rem;
    }
    button {
      background: transparent;
      border: 0.063rem solid rgba(0, 0, 0, 0.15);
      height: 2.375rem;
      width: 2.375rem;
    }
  }
  .icon-wrapper {
    z-index: 1;
  }

  .text {
    background: transparent;
    border: 0.063rem solid rgba(0, 0, 0, 0.15);
    height: 1.938rem;
    width: 1.938rem;
  }

  .modal-full {
    min-width: 100%;
    margin: 0;
    background: #ffffff;
}

 .modal-full .modal-content {
    height: 100vh;
    border: none;
    border-radius: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .modal-body {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .modal-dialog {
    position: fixed;
    margin: 0;
    width: 100%;
    height: 100%;
    padding: 0;
    background: #ffffff;
  }

  .modal-backdrop {
    display: none;
  }

  .modal {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    overflow: hidden;
  }

  .compare{
    text-decoration: underline;
  }
  .err-alert {
    position: fixed;
    bottom: 0;
    right: 7.063rem;
    height: 80vh;
    padding-bottom: 10%;
  }

  .exclam {
    width: 3rem; 
    height: 3rem; 
    padding-right: 0.5rem;
  }
  .inner-icons {
    margin-left: 0.5rem;
  }
  .inner-icons div {
    padding-right: 1rem;
    p{
      font-size: 0.625rem;
    }
    .embed {
      border-radius: 0 !important;
      margin-top: 0.5rem;
      margin-bottom: 0.3rem;
    }
  }
  @keyframes dot-keyframes {
    0% {
      opacity: .4;
      transform: scale(1, 1);
    }
  
    50% {
      opacity: 1;
      transform: scale(1.2, 1.2);
    }
  
    100% {
      opacity: .4;
      transform: scale(1, 1);
    }
  }
  
  .loading-dots {
    text-align: center;
    width: 100%;
    
    &--dot {
      animation: dot-keyframes 1.5s infinite ease-in-out;
      background-color: #096DD9;
      border-radius: 0.75rem;
      display: inline-block;
      height: 0.75rem;
      width: 0.75rem;
      margin-right: .5rem;
      
      &:nth-child(2) {
        animation-delay: .5s;
      }
      
      &:nth-child(3) {
        animation-delay: 1s;
      }
    }
  }
}
