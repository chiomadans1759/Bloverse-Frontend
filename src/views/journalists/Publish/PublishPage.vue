<template>
  <main id="publish-page">
    <nav class="nav container d-flex justify-content-between align-items-center publish-nav">
      <a class="nav-link" @click.prevent="goBack">
         <i class="fal fa-arrow-left back-icon"></i>
         <span class="text-muted ml-2">Back</span>
      </a>
      <p class="nav-link text-uppercase small" v-if="showThis">saving... to draft</p>
      <button class="btn btn-sm btn-primary-outline preview" v-if="showThis">Preview to post</button>
    </nav> 
     <div class="container">
       <div class="row">
        <div class="col-lg-1 icon-wrapper mt-4">
          <ul class="text-icons">
            <li>
              <button class="btn rounded-circle d-flex justify-content-center show" @click.prevent="showTextBox">
                <i class="fal fa-font"></i>
              </button>
            </li>
            <li>
              <Poptip placement="right">
                <div slot="content" class="d-flex justify-content-between inner-icons">
                  <div>
                    <button class="btn rounded-circle d-flex justify-content-center">
                      <i class="fal fa-arrow-from-bottom"></i>
                    </button>
                    <p>Upload</p>
                  </div>
                   <div>
                    <button class="btn rounded-circle d-flex justify-content-center">
                      <i class="fal fa-search"></i>
                    </button>
                  <p>Search</p>
                  </div>
                  <div>
                    <button class="btn rounded-circle d-flex justify-content-center">
                      <i class="far fa-th"></i>
                    </button>
                  <p>Multiple</p>
                  </div>
                </div>
                <button class="btn rounded-circle d-flex justify-content-center">
                  <i class="fal fa-image"></i>
                </button>
              </Poptip>
            </li>
            <li>
              <Poptip placement="right">
                <div slot="content" class="d-flex justify-content-between inner-icons">
                  <div>
                    <button class="btn rounded-circle d-flex justify-content-center">
                      <i class="fal fa-arrow-from-bottom"></i>
                    </button>
                    <p>Upload</p>
                  </div>
                   <div>
                    <button class="btn rounded-circle d-flex justify-content-center text">
                      <i class="fal fa-link"></i>
                    </button>
                  <p>Embedded</p>
                  </div>
                </div>
                <button class="btn rounded-circle d-flex justify-content-center">
                  <i class="fal fa-video"></i>
                </button>
              </Poptip>
              </li>
            <li>
              <Poptip placement="right">
                <div slot="content" class="d-flex justify-content-between inner-icons">
                  <div>
                    <input type="text" placeholder="Enter a url link" class="embed">
                    <p>Add a link from twitter, facebook</p>
                  </div>
                </div>
                <button class="btn rounded-circle d-flex justify-content-center">
                  <i class="fal fa-link"></i>
                </button>
              </Poptip>
              </li>
          </ul>
        </div>
        <div class="col-lg-6 mt-4">
          <form>
            <div class="form-group">
              <input type="text" class="form-control p-0 one" placeholder="Add a title" autofocus>
            </div>
            <div class="form-group">
              <input type="text" class="form-control p-0 two" placeholder="Add keypoint">
            </div>
            <div class="form-group pb-4">
              <textarea class="mytext" ref="autoText" v-model="textLength" v-if="textArea" @input="textareaResize"></textarea>
            </div>
          </form>
        </div>
        <div class="col-lg-3 err-alert" v-if="showMe">
          <div class="d-flex justify-content-end mb-4">
            <button class="btn rounded-circle d-flex justify-content-center text" @click.prevent="closeEntitySection">
              <i class="fal fa-arrow-right"></i>
            </button>
          </div>
          <error-alert>
            <p>We have 3 articles that have similar title as yours. <a class="compare">Compare</a></p>
          </error-alert>
          <h6 class="my-4">Entities <span class="badge badge-primary">50</span></h6>
          <entity-section></entity-section>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
import ErrorAlert from '@/components/ErrorAlert/ErrorAlert.vue'
import EntitySection from '@/components/EntitySection/EntitySection.vue'
import { Poptip } from 'iview'

export default {
  components: { ErrorAlert, Poptip, EntitySection },
  data() {
    return {
      textArea: false,
      showMe: false,
      textLength: "",
      showThis: false
    }
  },
  methods: {
    textareaResize() {
      this.$refs.autoText.style.minHeight = this.$refs.autoText.scrollHeight + 'px';
      if(this.textLength.length > 0) {
        this.showMe = true;
        this.showThis = true;
      } else {
        this.showMe = false;
        this.showThis = false;
      }
      
    },
    showTextBox() {
      this.textArea = true;
      this.$nextTick(() => this.$refs.autoText.focus())
    },
    closeEntitySection() {
      this.showMe = false
    },
    goBack() {
      this.$router.push({ path: 'creators/profile' });
    }
  }
}
</script>

<style lang="scss" scoped>
@import "./PublishPage"
</style>
