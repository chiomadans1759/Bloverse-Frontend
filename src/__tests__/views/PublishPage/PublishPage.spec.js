import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import PublishPage from '@/views/journalists/Publish/PublishPage.vue';

describe('PublishPage Component', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(PublishPage, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should increase the `textArea`', () => {
    const wrapper = shallowMount(PublishPage, {
      stubs: { RouterLink: RouterLinkStub }
    });
    wrapper.find('.show').trigger('click');

    expect(wrapper.vm.textArea).toBe(true);
  });

  it('should display `showMe`', () => {
    const wrapper = shallowMount(PublishPage, {
      stubs: 
        { RouterLink: RouterLinkStub }
    });
    wrapper.find('.text').trigger('click');

    expect(wrapper.vm.showMe).toBe(false);
  });

  it('should display `showThis`', () => {
    const wrapper = shallowMount(PublishPage, {
      stubs: 
      { RouterLink: RouterLinkStub }
    });
    wrapper.find('.btn').trigger('input');
    wrapper.find('.nav-link').trigger('input');

    expect(wrapper.vm.showThis).toBe(false);
  })
})
