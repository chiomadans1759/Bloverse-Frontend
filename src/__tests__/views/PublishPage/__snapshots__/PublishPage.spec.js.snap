// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishPage Component should render correctly 1`] = `
<main
  id="publish-page"
>
  <nav
    class="nav container d-flex justify-content-between align-items-center publish-nav"
  >
    <a
      class="nav-link"
    >
      <i
        class="fal fa-arrow-left back-icon"
      />
       
      <span
        class="text-muted ml-2"
      >
        Back
      </span>
    </a>
     
    <!---->
     
    <!---->
  </nav>
   
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-1 icon-wrapper mt-4"
      >
        <ul
          class="text-icons"
        >
          <li>
            <button
              class="btn rounded-circle d-flex justify-content-center show"
            >
              <i
                class="fal fa-font"
              />
            </button>
          </li>
           
          <li>
            <div
              class="ivu-poptip"
            >
              <div
                class="ivu-poptip-rel"
              >
                 
                <button
                  class="btn rounded-circle d-flex justify-content-center"
                >
                  <i
                    class="fal fa-image"
                  />
                </button>
              </div>
               
              <div
                class="ivu-poptip-popper"
                name="fade"
                style="display: none;"
              >
                <div
                  class="ivu-poptip-content"
                >
                  <div
                    class="ivu-poptip-arrow"
                  />
                   
                  <!---->
                   
                  <div
                    class="ivu-poptip-inner"
                  >
                    <div
                      class="ivu-poptip-title"
                    >
                      <div
                        class="ivu-poptip-title-inner"
                      >
                        
                      </div>
                    </div>
                     
                    <div
                      class="ivu-poptip-body"
                    >
                      <div
                        class="ivu-poptip-body-content"
                      >
                        <div
                          class="d-flex justify-content-between inner-icons"
                        >
                          <div>
                            <button
                              class="btn rounded-circle d-flex justify-content-center"
                            >
                              <i
                                class="fal fa-arrow-from-bottom"
                              />
                            </button>
                             
                            <p>
                              Upload
                            </p>
                          </div>
                           
                          <div>
                            <button
                              class="btn rounded-circle d-flex justify-content-center"
                            >
                              <i
                                class="fal fa-search"
                              />
                            </button>
                             
                            <p>
                              Search
                            </p>
                          </div>
                           
                          <div>
                            <button
                              class="btn rounded-circle d-flex justify-content-center"
                            >
                              <i
                                class="far fa-th"
                              />
                            </button>
                             
                            <p>
                              Multiple
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
           
          <li>
            <div
              class="ivu-poptip"
            >
              <div
                class="ivu-poptip-rel"
              >
                 
                <button
                  class="btn rounded-circle d-flex justify-content-center"
                >
                  <i
                    class="fal fa-video"
                  />
                </button>
              </div>
               
              <div
                class="ivu-poptip-popper"
                name="fade"
                style="display: none;"
              >
                <div
                  class="ivu-poptip-content"
                >
                  <div
                    class="ivu-poptip-arrow"
                  />
                   
                  <!---->
                   
                  <div
                    class="ivu-poptip-inner"
                  >
                    <div
                      class="ivu-poptip-title"
                    >
                      <div
                        class="ivu-poptip-title-inner"
                      >
                        
                      </div>
                    </div>
                     
                    <div
                      class="ivu-poptip-body"
                    >
                      <div
                        class="ivu-poptip-body-content"
                      >
                        <div
                          class="d-flex justify-content-between inner-icons"
                        >
                          <div>
                            <button
                              class="btn rounded-circle d-flex justify-content-center"
                            >
                              <i
                                class="fal fa-arrow-from-bottom"
                              />
                            </button>
                             
                            <p>
                              Upload
                            </p>
                          </div>
                           
                          <div>
                            <button
                              class="btn rounded-circle d-flex justify-content-center text"
                            >
                              <i
                                class="fal fa-link"
                              />
                            </button>
                             
                            <p>
                              Embedded
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
           
          <li>
            <div
              class="ivu-poptip"
            >
              <div
                class="ivu-poptip-rel"
              >
                 
                <button
                  class="btn rounded-circle d-flex justify-content-center"
                >
                  <i
                    class="fal fa-link"
                  />
                </button>
              </div>
               
              <div
                class="ivu-poptip-popper"
                name="fade"
                style="display: none;"
              >
                <div
                  class="ivu-poptip-content"
                >
                  <div
                    class="ivu-poptip-arrow"
                  />
                   
                  <!---->
                   
                  <div
                    class="ivu-poptip-inner"
                  >
                    <div
                      class="ivu-poptip-title"
                    >
                      <div
                        class="ivu-poptip-title-inner"
                      >
                        
                      </div>
                    </div>
                     
                    <div
                      class="ivu-poptip-body"
                    >
                      <div
                        class="ivu-poptip-body-content"
                      >
                        <div
                          class="d-flex justify-content-between inner-icons"
                        >
                          <div>
                            <input
                              class="embed"
                              placeholder="Enter a url link"
                              type="text"
                            />
                             
                            <p>
                              Add a link from twitter, facebook
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
       
      <div
        class="col-lg-6 mt-4"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              autofocus="autofocus"
              class="form-control p-0 one"
              placeholder="Add a title"
              type="text"
            />
          </div>
           
          <div
            class="form-group d-flex"
          >
            <input
              class="form-control p-0 two"
              placeholder="Add keypoint"
              type="text"
            />
             
            <div>
              <span>
                <button
                  class="btn rounded-circle d-flex justify-content-center ml-3 text"
                >
                  <i
                    class="fal fa-plus"
                  />
                </button>
              </span>
            </div>
          </div>
           
          <div
            class="form-group pb-4"
          >
            <!---->
          </div>
        </form>
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    aria-hidden="true"
    class="modal"
    id="previewModal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-full"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-body"
        >
          <div>
            <div
              class="loading-dots"
            >
              <div
                class="loading-dots--dot"
              />
               
              <div
                class="loading-dots--dot"
              />
               
              <div
                class="loading-dots--dot"
              />
            </div>
             
            <p>
              Scanning Article For Plagiarism...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
